IndustrialUnitProvider/UnitMapper.cs

```diff
@@ -0,0 +1,99 @@
+﻿using IndustrialUnit.Model.Model;
+using IndustrialUnitDatabase;
+using OfficeOpenXml;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace IndustrialUnitProvider
+{
+  public class UnitMapper
+  {
+    public void LoadUnitsFromSheet(string file, ref List<string> logMessage)
+    {
+      List<Equipment> equipments = new List<Equipment>();
+      var sheetEquipment = ExcelWorker.ReadExcel(file, ValidSheetNames.Equipment.ToString(), ref logMessage);
+      if(sheetEquipment != null)
+      AssignValue(equipments, sheetEquipment, ref logMessage);
+
+      List<Valve> valves = new List<Valve>();
+      var sheetValve = ExcelWorker.ReadExcel(file, ValidSheetNames.Valve.ToString(), ref logMessage);
+      if(sheetValve != null)
+      AssignValue(valves, sheetValve, ref logMessage);
+
+      List<Instrument> instruments = new List<Instrument>();
+      var sheetInstruments = ExcelWorker.ReadExcel(file, ValidSheetNames.Instrument.ToString(), ref logMessage);
+      if(sheetInstruments != null)
+      AssignValue(instruments, sheetInstruments, ref logMessage);
+    }
+
+    public void AssignValue<T>(List<T> parameterCollection, ExcelWorksheet sheet, ref List<string> logMessage) where T : class, new()
+    {
+      PropertyInfo[] properties = typeof(T).GetProperties();
+
+      var validation = new ExcelValidation();
+      var columnNameToIndex = validation.CollectColumnNamesFromExcel(sheet);
+
+      if (!validation.ValidateColumnNames(columnNameToIndex, properties, sheet, ref logMessage))  return;
+
+      for (int rowIndex = 2; rowIndex < sheet.Dimension.Rows + 1; rowIndex++)
```

Comment:
Ha percekig fut, érdemes ide tenni egy breakpointot, és megnézni,  hogy a `sheet.Dimension.Rows` valójában mit ad vissza, mert lehet, hogy van a végén pár tízezer üres sor, amit beleszámol. Lehet, hogy elég lenne megállni, amint találsz egy üres sort, és felvenni a specifikációba, hogy nem lehetnek üres sorok az excelben (ez elég gyakori pont az ilyen problémák miatt, amikor az excel úgy menti el a fájlt, hogy 65000 sor van benne, pedig valójában csak a töredéke tartalmaz adatot).

Reply:
Valóban minden sort megszámol, de csak azokat teszi be a belső listába ahol egy bizonyos cellában érték van, de fut egy kört azon is amiben nincs semmi. Viszont ez a kigyüjtés gyors, néhány másodperc. A húzós az SQL transaction ott kerreg a  HDD sokáig. 
SSD-n 2perc, 5éves HDD-n 20 perc. Készítettem egy progress bart hozzá, user kivárja. :D